server/data/data-library/src/main/java/org/apache/james/user/lib/model/DefaultUser.java

```diff
@@ -86,7 +94,9 @@ public boolean verifyPassword(String pass) {
     @Override
     public boolean setPassword(String newPass) {
         try {
-            hashedPassword = DigestUtil.digestString(newPass, algorithm);
+            String newCredentials = preferredAlgorithm.isSalted() ? userName.asString() + newPass : newPass;
```

Comment:
I don't think when `salt` is a `username` value will make sense. I thought it should be a random string, that only the server site knows.

Reply:
See above.